mlrun/projects/pipelines.py

```diff
@@ -820,8 +820,9 @@ def run(
         namespace=None,
         source=None,
     ) -> typing.Optional[_PipelineRunStatus]:
-        workflow_name = name.split("-")[-1] if f"{project.name}-" in name else name
-
+        workflow_name = (
+            name.replace(f"{project.name}", "")[1:] if project.name in name else name
+        )
```

Comment:
```suggestion
        )workflow_name =  name.replace(project.name, "")[1:] if project.name in name else name
```

if you don't use f"{}-" with the dash then the f-string is redundant.
if you assume the project name with the dash is at the beginning of the string, you can either remove the f-string, use it with the dash inside, or maybe use something like: 
name[len(project.name) + 1:] if project.name in name else name

Reply:
actually I'm noticing now following Tomer's comment, what if:
project.name = "test"
name = "test"

after we added the project_name it will be "test-test", and your replace will just leave it empty.
in any case where for some reason `name` contain the `project.name`, it will be removed (imagine if the project name is "a" or something).
I would replace it with:
```

 workflow_name = (
            name.lstrip(project.name).lstrip("-") if project.name in name else name
        )
```
spilt to two lstrips based on our offline conversation :)
and maybe add on the way some more cases to the test?